tests/components/vevor_heater/test_sensor.py

```diff
@@ -0,0 +1,223 @@
+                )
+                _assert_state(state, expected_value=value, expected_unit=unit)
+            assert await hass.config_entries.async_unload(entry.entry_id)
+            await hass.async_block_till_done()
+
+            pass
+
+
+async def test_sensors_off(enable_bluetooth: None, hass: HomeAssistant) -> None:
+    """Tests sensors when the heater is off."""
+    status_to_set = VevorHeaterStatus(power_status=PowerStatus.OFF)
+
+    with patch(
+        "homeassistant.components.bluetooth.async_ble_device_from_address",
+        return_value=FIXTURE_BLE_DEVICE,
+    ):
+
+        def update_self_status(self, _):
+            self.status = status_to_set
+
+        with patch.object(
+            VevorDevice, "refresh_status", autospec=True
+        ) as refresh_status:
+            refresh_status.side_effect = update_self_status
+
+            entry = MockConfigEntry(
+                domain=DOMAIN, unique_id="Foo", data={"address": "AB:CD:EF:01:23:45"}
+            )
+            entry.add_to_hass(hass)
+
+            assert await hass.config_entries.async_setup(entry.entry_id)
+
+            await hass.async_block_till_done()
+            status_to_set = VevorHeaterStatus(power_status=PowerStatus.OFF)
+
+            async_fire_time_changed(hass, dt_util.utcnow() + DEFAULT_UPDATE_INTERVAL)
+            await hass.async_block_till_done()
+
+            for sensor, value, unit in [
+                ("operational_mode", "unknown", None),
+                ("operational_status", "unknown", None),
+                ("source_voltage", "unknown", "V"),
+                ("elevation", "unknown", "m"),
+                ("target_temperature", "unknown", "°C"),
+                ("target_power_level", "unknown", None),
+                ("current_power_level", "unknown", None),
+                ("combustion_chamber_temperature", "unknown", "°C"),
+                ("room_temperature", "unknown", "°C"),
+                ("error", "unknown", None),
+            ]:
+                state = hass.states.get(
+                    f"sensor.vevor_{FIXTURE_BLE_DEVICE.name.lower()}_{sensor}"
+                )
+                _assert_state(state, expected_value=value, expected_unit=unit)
+            assert await hass.config_entries.async_unload(entry.entry_id)
+            await hass.async_block_till_done()
+
+
+def _assert_state(
+    state: Any, expected_value: Any, expected_unit: Optional[str]
+) -> None:
+    if expected_value is None:
+        # If we expect the state not to be present it can't have any unit.
+        # Failing assert indicates a misconfiguration of the test, not a SUT malfunction.
+        assert expected_unit is None
+
+        # Now check the SUT
+        assert state is None
+    else:
+        assert state.state == expected_value
+        if expected_unit is None:
+            assert "unit" not in state.attributes
+        else:
+            assert state.attributes["unit_of_measurement"] == expected_unit
```

Comment:
We don't want branching in tests as it makes them harder to maintain, review, and follow.

There are usually two options to avoid this:

1. Break the test up into two tests
2. use `@pytest.mark.parametrize` (lots of examples in the code base)

Reply:
This is not a branching test per se, it's just a helper method to evaluate the presence / absence of a state according to the expectation. Similar logic can be found e.g. in assert methods of `unittest.testcase`.